### ISSUE TITLE
Internal cross-validation

### ISSUE THREAD

**Issue by jernsting** [automl/auto-sklearn#711]
I am a bit confused about which technique of cross-validation you are using internally.

As far as I understand it, the algorithm shown in the paper works iteratively (Figure 1) [here](https://papers.nips.cc/paper/5872-efficient-and-robust-automated-machine-learning). But on which set is the pipeline proposed by the algorithm evaluated? Is this the mentioned holdout-iterative-fit, which is set as  resampling_strategy or is this something else?

**Comment by ballcap231** [comment]
I'm curious what even holdout-iterative-fit/ cross-validation iterative fit (partial-cv) means.

**Comment by mfeurer** [comment]
The data given by the user is used to both train and validate the machine learning pipeline. How the data is split depends on the resampling strategy. The default one is holdout, and cross-validation is popular, too. These are simply the splitters from scikit-learn applied to the whole dataset. If you cant to know the test performance, you need to keep a test set aside yourself.

Section 6 of our Auto-sklearn paper used the holdout strategy. Sections 4 & 5 used the `holdout-iterative-fit`, which is a bad name for a procedure called intensification, described in [Algorithm 2 of Hutter et al., 2011](https://ml.informatik.uni-freiburg.de/papers/11-LION5-SMAC.pdf). Basically, this is cross-validation, but folds are only 'executed', if there is a chance of improving over the currently best known configuration.

Please reopen if you have any further questions.
